library/Data/Field.hs

```diff
@@ -116,8 +135,25 @@ instance Field f => Num (AField f) where
             go n | even n = two `mul` fromInteger (n `Prelude.div` 2)
             go n          = one `add` fromInteger (n - 1)
 
-instance Show f => Show (AField f) where
-    show = show . unAField
+instance TextField f => TextField (AField f) where
+    parseField = AField <$> parseField
+    showField = showField . unAField
+
+instance TextField f => Show (AField f) where
+    show = showField . unAField
+
+-- Note that the parser for @Rational@ accepts ONLY plain integers. This is because we pretty-print
+-- rationals as @a / b@ and so @/@ is parsed elsewhere as regular division.
+instance TextField Rational where
+    parseField = asum
+        [ try (div <$> parseFieldDefault <* symbol "/" <*> parseFieldDefault)
+        , parseFieldDefault
+        , parens parseField
+        ]
+
```

Comment:
This looks OK. It does allow the slightly odd-looking `-5/-3`, but I think that's perfectly reasonable.  I see now that Haskell does deal with `read "-5 % -3"` successfully (but not `-5 / -3` as an expression), and the earlier parser wouldn't have accepted that.  Hwoever this is definitely an improvement on trying to use the Haskell syntax for rationals, which no-one else uses anyway.

Reply:
Great, thanks.